src/DurableTask.Generators/AzureFunctions/DurableFunction.cs

```diff
@@ -21,59 +21,79 @@ public class DurableFunction
         public string Name { get; }
         public DurableFunctionKind Kind { get; }
         public TypedParameter Parameter { get; }
-        public TypeSyntax ReturnType { get; }
+        public string ReturnType { get; }
 
-        public DurableFunction(string fullTypeName, string name, DurableFunctionKind kind, TypedParameter parameter, TypeSyntax returnTypeSyntax, HashSet<string> requiredNamespaces)
+        public DurableFunction(
+            string fullTypeName,
+            string name,
+            DurableFunctionKind kind,
+            TypedParameter parameter,
+            ITypeSymbol returnType,
+            HashSet<string> requiredNamespaces)
         {
             this.FullTypeName = fullTypeName;
             this.RequiredNamespaces = requiredNamespaces;
             this.Name = name;
             this.Kind = kind;
             this.Parameter = parameter;
-            this.ReturnType = returnTypeSyntax;
+            this.ReturnType = SyntaxNodeUtility.GetRenderedTypeExpression(returnType, false);
         }
 
         public static bool TryParse(SemanticModel model, MethodDeclarationSyntax method, out DurableFunction? function)
         {
-            function = null;
-
             if (!SyntaxNodeUtility.TryGetFunctionName(model, method, out string? name) || name == null)
             {
+                function = null;
                 return false;
             }
 
             if (!SyntaxNodeUtility.TryGetFunctionKind(method, out DurableFunctionKind kind))
             {
+                function = null;
                 return false;
             }
 
             if (!SyntaxNodeUtility.TryGetReturnType(method, out TypeSyntax returnType))
             {
+                function = null;
                 return false;
             }
 
-            if (!SyntaxNodeUtility.TryGetParameter(method, kind, out TypedParameter? parameter) || parameter == null)
+            INamedTypeSymbol taskSymbol = model.Compilation.GetTypeByMetadataName("System.Threading.Tasks.Task`1")!;
+            INamedTypeSymbol returnSymbol = (INamedTypeSymbol)model.GetTypeInfo(returnType).Type!;
+            if (SymbolEqualityComparer.Default.Equals(returnSymbol.OriginalDefinition, taskSymbol))
+            {
+                // this is a Task<T> return value, lets pull out the generic.
+                returnSymbol = (INamedTypeSymbol)returnSymbol.TypeArguments[0];
+            }
+
+            if (!SyntaxNodeUtility.TryGetParameter(model, method, kind, out TypedParameter? parameter) || parameter == null)
             {
+                function = null;
                 return false;
             }
 
             if (!SyntaxNodeUtility.TryGetQualifiedTypeName(model, method, out string? fullTypeName))
             {
+                function = null;
                 return false;
             }
```

Comment:
Would it make sense to move the `TryGetParameter` and `TryGetQualifiedName` checks further up, above the new return type code you added? Could that short-circuiting help save some CPU cycles or do we very rarely expect those methods to return false?